src/absil/ilwrite.fs

```diff
@@ -439,15 +440,40 @@ type MethodDefKey(tidx: int, garity: int, nm: string, rty: ILType, argtys: ILTyp
     member key.IsStatic = isStatic
     override x.GetHashCode() = hashCode
     override x.Equals(obj: obj) = 
-        match obj with 
-        | :? MethodDefKey as y -> 
-            tidx = y.TypeIdx && 
-            garity = y.GenericArity && 
-            nm = y.Name && 
-            // note: these next two use structural equality on AbstractIL ILType values
-            rty = y.ReturnType && 
-            List.lengthsEqAndForall2 (fun a b -> a = b) argtys y.ArgTypes &&
-            isStatic = y.IsStatic
+        match obj with
+        | :? MethodDefKey as y ->
+                let compareILTypes o1 o2 =
+                     let getScopeAssemblyRef (o:ILType) =
+                         match o with
+                         | ILType.Value v ->
+                             match v.Scope with
+                             | ILScopeRef.PrimaryAssembly -> Some (ilg.primaryAssemblyRef.QualifiedName)
+                             | ILScopeRef.Assembly aref -> Some (aref.QualifiedName)
+                             | _ -> None
+                         | _ -> None
+
+                     let scope1 = getScopeAssemblyRef o1
+                     let scope2 = getScopeAssemblyRef o2
+                     match scope1, scope2 with
+                     | Some s1, Some s2 ->
```

Comment:
Handling this case doesn't check the types themselves but only their scopes.

Reply:
Before, we were doing structural equality on ILType. But because we want to have equivalency for ILScope.PrimaryAssembly based on what is in `ILGlobals`, we might have to right a custom equality function that goes through the `ILType`. Another possible way to do this were to rescope ILType to replace the ILScopeRef.Primary with a ILScopeRef.Assembly.